sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs

```diff
@@ -641,8 +641,8 @@ public static partial class ComputeExtensions
         public static Azure.Pageable<Azure.ResourceManager.Compute.Models.RunCommandDocumentBase> GetVirtualMachineRunCommands(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.Core.AzureLocation location, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public static Azure.AsyncPageable<Azure.ResourceManager.Compute.Models.RunCommandDocumentBase> GetVirtualMachineRunCommandsAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, Azure.Core.AzureLocation location, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public static Azure.ResourceManager.Compute.VirtualMachineCollection GetVirtualMachines(this Azure.ResourceManager.Resources.ResourceGroupResource resourceGroupResource) { throw null; }
-        public static Azure.Pageable<Azure.ResourceManager.Compute.VirtualMachineResource> GetVirtualMachines(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, string statusOnly = null, string filter = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public static Azure.AsyncPageable<Azure.ResourceManager.Compute.VirtualMachineResource> GetVirtualMachinesAsync(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, string statusOnly = null, string filter = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public static Azure.Pageable<Azure.ResourceManager.Compute.VirtualMachineResource> GetVirtualMachines(this Azure.ResourceManager.Resources.SubscriptionResource subscriptionResource, string statusOnly = null, string filter = null, Azure.ResourceManager.Compute.Models.ExpandTypesForListVm? expand = default(Azure.ResourceManager.Compute.Models.ExpandTypesForListVm?), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
Member 'public Azure.Pageable<Azure.ResourceManager.Compute.VirtualMachineResource> Azure.ResourceManager.Compute.ComputeExtensions.GetVirtualMachines(Azure.ResourceManager.Resources.SubscriptionResource, System.String, System.String, System.Threading.CancellationToken)' does not exist in the implementation but it does exist in the contract.	Azure.ResourceManager.Compute	

I get above build error. how do i resolve this? the method didnt really change. do i still need to create a customization method because the new parameter comes before cancellationToken?

Reply:
Because we are adding a new optional parameter to this operation, and this is totally allowed in swagger, but here, since we have the cancellationToken, we could never put the new optional parameter at the end of the list, therefore it will always be a breaking change.

To resolve this, you will have to create a partial class of this type, put the old method there and calling the new one in its implementation. To make sure the code could compile, you need to make all the parameters in the "old method" required, something like this
```
public partial class ComputeExtensions // we do not need to create a new file if this file always exists in the customization or custom directory
{
public static Azure.Pageable<VirtualMachineResource> GetVirtualMachines(this SubscriptionResource subscriptionResource, string statusOnly, string filter, CancellationToken cancellationToken)
{
return subscriptionResource.GetVirtualMachines(statusOnly, filter, null, cancellationToken);
}
}
```